pkg/component/gardenercontrollermanager/configmaps.go

```diff
@@ -0,0 +1,168 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gardenercontrollermanager
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	componentbaseconfigv1alpha1 "k8s.io/component-base/config/v1alpha1"
+	"k8s.io/utils/pointer"
+
+	controllermanagerv1alpha1 "github.com/gardener/gardener/pkg/controllermanager/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/logger"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+const (
+	configMapControllerManagerPrefix = "gardener-controller-manager-config"
+	dataConfigKey                    = "config.yaml"
+)
+
+var controllerManagerCodec runtime.Codec
+
+func init() {
+	controllerManagerScheme := runtime.NewScheme()
+	utilruntime.Must(controllermanagerv1alpha1.AddToScheme(controllerManagerScheme))
+
+	var (
+		ser = json.NewSerializerWithOptions(json.DefaultMetaFactory, controllerManagerScheme, controllerManagerScheme, json.SerializerOptions{
+			Yaml:   true,
+			Pretty: false,
+			Strict: false,
+		})
+		versions = schema.GroupVersions([]schema.GroupVersion{
+			controllermanagerv1alpha1.SchemeGroupVersion,
+		})
+	)
+
+	controllerManagerCodec = serializer.NewCodecFactory(controllerManagerScheme).CodecForVersions(ser, ser, versions, versions)
+}
+
+func (g *gardenerControllerManager) configMapControllerManagerConfig() (*corev1.ConfigMap, error) {
+	controllerManagerConfig := &controllermanagerv1alpha1.ControllerManagerConfiguration{
+		GardenClientConnection: componentbaseconfigv1alpha1.ClientConnectionConfiguration{
+			QPS:        100,
+			Burst:      130,
+			Kubeconfig: gardenerutils.PathGenericKubeconfig,
+		},
+		Controllers: controllermanagerv1alpha1.ControllerManagerControllerConfiguration{
+			CertificateSigningRequest: &controllermanagerv1alpha1.CertificateSigningRequestControllerConfiguration{
+				ConcurrentSyncs: pointer.Int(5),
+			},
+			ControllerRegistration: &controllermanagerv1alpha1.ControllerRegistrationControllerConfiguration{
+				ConcurrentSyncs: pointer.Int(20),
+			},
+			Project: &controllermanagerv1alpha1.ProjectControllerConfiguration{
+				ConcurrentSyncs: pointer.Int(20),
+				Quotas:          g.values.Quotas,
+			},
+			SecretBinding: &controllermanagerv1alpha1.SecretBindingControllerConfiguration{
+				ConcurrentSyncs: pointer.Int(20),
```

Comment:
Why are the concurrent syncs for some controllers changed from what is present in the current charts, but not others?

Reply:
The chart provides "default" values, but here we use "best practice" values from real-world experience of running Gardener on large landscapes. We could also adapt these values in the `values.yaml` to reflect the same, or we just leave it since productive Gardener installations should be managed by `gardener-operator` in the long run. 🤷🏻

Reply:
Let's leave it as is for now and let `gardener-operator` take over in the long run.